src/CartesianProduct.jl

```diff
@@ -272,3 +308,73 @@ function ∈(x::AbstractVector{N}, cpa::CartesianProductArray{N, <:LazySet{N}}
     end
     return true
 end
+
+"""
+    vertices_list(cpa::CartesianProductArray{N})::Vector{Vector{N}} where N<:Real
+
+Return the list of vertices of a (polytopic) Cartesian product.
+
+### Input
+
+- `cpa` -- Cartesian product
+
+### Output
+
+A list of vertices.
+
+### Algorithm
+
+We assume that the underlying sets are polytopic.
+Then the high-dimensional set of vertices is just the Cartesian product of the
+low-dimensional sets of vertices.
+"""
+function vertices_list(cpa::CartesianProductArray{N})::Vector{Vector{N}} where N<:Real
+    # collect low-dimensional vertices lists
+    vlist_low = [vertices_list(X) for X in array(cpa)]
+
+    # create high-dimensional vertices list
+    m = 1
+    for vl in vlist_low
+        m *= length(vl)
+    end
+    vlist = Vector{Vector{N}}(undef, m)
+    indices = ones(Int, length(vlist_low))
+    indices_max = [length(vl) for vl in vlist_low]
+    v = zeros(N, dim(cpa))
+    j = 1
+    for vl in vlist_low
+        v_low = vl[1]
+        v[j:j+length(v_low)-1] = v_low
+        j += 1
```

Comment:
i don't get this one. is `j += 1` correct?

Reply:
`j` is the block's index. I walk through all blocks until I find a block where `indices[j] < indices_max[j]`.

Reply:
Fixed!